synthesizer/src/vm/helpers/rewards.rs

```diff
@@ -51,27 +51,46 @@ pub fn staking_rewards<N: Network>(
         return stakers.clone();
     }
 
+    // First, validate all validators and create a map of valid validators.
+    let valid_validators: IndexMap<_, _> = committee
+        .members()
+        .iter()
+        .filter_map(|(validator, (validator_stake, _is_open, commission_rate))| {
+            // Skip validators with invalid commission rates.
+            if *commission_rate > 100 {
+                error!("Commission rate ({commission_rate}) is greater than 100 - skipping validator {validator}");
+                return None;
+            }
+            // Skip validators with more than 25% of total stake.
+            if *validator_stake > committee.total_stake().saturating_div(4) {
+                trace!("Validator {validator} has more than 25% of the total stake - skipping all stakers");
+                return None;
+            }
+            Some((*validator, (*validator_stake, *commission_rate)))
+        })
+        .collect();
+
+    // Track validators not in committee.
+    // Pre-allocating with an expected capacity prevents reallocation while the mutex is held.
+    let hashset_capacity = committee.members().len();
+    let missing_validators =
+        std::sync::Mutex::new(std::collections::HashSet::<Address<N>>::with_capacity(hashset_capacity));
+
     // Compute the updated stakers.
     cfg_iter!(stakers)
         .map(|(staker, (validator, stake))| {
-            // If the validator is not in the committee, skip the staker.
-            let Some((validator_stake, _is_open, commission_rate)) = committee.members().get(validator) else {
-                trace!("Validator {validator} is not in the committee - skipping {staker}");
+            // If the validator is not in the valid validators list, skip the staker.
+            let Some((validator_stake, commission_rate)) = valid_validators.get(validator) else {
+                // Log validator not in committe.
```

Comment:
Can you disable this block if we are at the default verbosity level (which is Debug IIRC)?

You can use [log::log_enabled!](https://docs.rs/log/latest/log/macro.log_enabled.html) for this.

Reply:
(Sorry, I put the comment at the wrong location before. It seems like this is the only part we can skip.)

Reply:
I don't see `log_enabled` used anywhere else in snarkVM 🤔 If you think we need it maybe you can make a new issue for it?